modules/codegen/src/main/scala/com/twilio/guardrail/generators/Http4sServerGenerator.scala

```diff
@@ -636,7 +636,7 @@ object Http4sServerGenerator {
             }
             if (!List("Unit", "Boolean", "Double", "Float", "Short", "Int", "Long", "Char", "String").contains(elemType.toString())) {
               val queryParamDecoder = q"""
-                implicit val ${Pat.Var(Term.Name(s"${argName.value}QueryParamDecoder"))}: QueryParamDecoder[$elemType] = (value: QueryParameterValue) =>
+                implicit val ${Pat.Var(Term.Name(s"${elemType}QueryParamDecoder"))}: QueryParamDecoder[$elemType] = (value: QueryParameterValue) =>
```

Comment:
These `elemType`s are not guaranteed to be fully qualified, as we won't always know that information, consider:

```
- name: foo
  x-scala-type: Foo
- name: foo2
  x-scala-type: com.example.Foo
```
given `imports=List("com.example._")` in build.sbt.

I wonder if
```
implicit def decoderDecoder[A: Decoder]: QueryParamDecoder[A] =
  Json.fromString(value.value).as[A]
    .leftMap(t => ParseFailure("Query decoding failed", t.getMessage))
    .toValidatedNel
```
would be more stable, relying on scalac to do the wiring up for us.

One unfortunate aspect here is that `Json.fromString` prevents non-string query parameters. I'd expect
```
- name: numeric
  type: integer
  format: int64
```
would fail to parse anything, see #184 and #292. This is outside of the scope of your library upgrade here, just raising awareness of an issue.